File: jsParser/parser/Component_clause1Visitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Component_clause1 = require('../domain/Component_clause1');

const Type_prefixVisitor = require('./Type_prefixVisitor');
const Type_specifierVisitor = require('./Type_specifierVisitor');
const Component_declaration1Visitor = require('./Component_declaration1Visitor');

class Component_clause1Visitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitComponent_clause1(ctx) {
        var type_prefix = null;
        var type_specifier = null;
        var component_declaration1 = null;

        if (ctx.type_prefix()) {
            var type_prefixVisitor = new Type_prefixVisitor.Type_prefixVisitor();
            type_prefix = type_prefixVisitor.visitType_prefix(ctx.type_prefix());
        }
        if (ctx.type_specifier()) {    
            const type_specifierVisitor = new Type_specifierVisitor.Type_specifierVisitor();
            type_specifier = type_specifierVisitor.visitType_specifier(ctx.type_specifier());
        }
        if (ctx.component_declaration1()) {
            var component_declaration1Visitor = new Component_declaration1Visitor.Component_declaration1Visitor();
            component_declaration1 = component_declaration1Visitor.visitComponent_declaration1(ctx.component_declaration1());
        }
        return new Component_clause1.Component_clause1(type_prefix, type_specifier, component_declaration1);
    }
};

Component_clause1Visitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitComponent_clause1 = this.visitComponent_clause1;
exports.Component_clause1Visitor = Component_clause1Visitor;

File: java/parser/src/main/java/gov/lbl/parser/parser/Component_clause1Visitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Component_clause1;
import gov.lbl.parser.domain.Component_declaration1;
import gov.lbl.parser.domain.Type_specifier;

public class Component_clause1Visitor extends modelicaBaseVisitor<Component_clause1> {
    @Override
    public Component_clause1 visitComponent_clause1(modelicaParser.Component_clause1Context ctx) {
        Type_prefixVisitor type_prefixVisitor = new Type_prefixVisitor();
        String type_prefix = ctx.type_prefix().accept(type_prefixVisitor);

        Type_specifierVisitor type_specifierVisitor = new Type_specifierVisitor();
        Type_specifier type_specifier = ctx.type_specifier().accept(type_specifierVisitor);

        Component_declaration1Visitor component_declaration1Visitor = new Component_declaration1Visitor();
        Component_declaration1 component_declaration1 = ctx.component_declaration1().accept(component_declaration1Visitor);
        
        return new Component_clause1(type_prefix, type_specifier, component_declaration1);
    }
}